codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +275,12 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                String http1Request = in.toString(0, 1024, CharsetUtil.US_ASCII);
```

Comment:
why 1024 ? Also is it possible that the buffer may have less readable bytes then 1024 ? If so should this be `in.toString(in.readerIndex(), Math.min(in.readableBytes(), 1024))` ?